src/PactNet.Native/PactExtensions.cs

```diff
@@ -51,17 +51,51 @@ public static IPactBuilderV3 UsingNativeBackend(this IPactV3 pact, int? port = n
         }
 
         /// <summary>
-        /// Initialise a new pact on the server with the correct version
+        /// Establish a new messagePact using the native backend
+        /// </summary>
+        /// <param name="messagePact">Pact details</param>
+        /// <returns>Pact builder</returns>
+        /// <remarks>
+        /// If multiple mock servers are started at the same time, you must make sure you don't supply the same port twice.
+        /// It is advised that the port is not specified whenever possible to allow PactNet to allocate a port dynamically
+        /// and ensure there are no port clashes
+        /// </remarks>
+        public static IMessagePactBuilderV3 UsingNativeBackend(this IMessagePactV3 messagePact)
+        {
+            NativeMockServer server = new NativeMockServer();
+            MessagePactHandle handle = InitialiseMessage(server, messagePact, PactSpecification.V3);
+
+            var builder = new NativeMessagePactBuilder(server, handle, messagePact.Config);
+            return builder;
+        }
+
+        /// <summary>
+        /// Initialise a new messagePact on the server with the correct version
         /// </summary>
         /// <param name="pact">Pact details</param>
         /// <param name="server">Server</param>
         /// <param name="version">Spec version</param>
-        /// <returns>Initialised pact handle</returns>
+        /// <returns>Initialised messagePact handle</returns>
         private static PactHandle InitialiseServer(NativeMockServer server, IPact pact, PactSpecification version)
         {
             PactHandle handle = server.NewPact(pact.Consumer, pact.Provider);
             server.WithSpecification(handle, version);
             return handle;
         }
+
+        /// <summary>
+        /// Initialise a new message messagePact with the correct version
+        /// </summary>
+        /// <param name="messagePact">Pact details</param>
+        /// <param name="server">Server</param>
+        /// <param name="version">Spec version</param>
+        /// <returns>Initialised message messagePact handle</returns>
+        private static MessagePactHandle InitialiseMessage(NativeMockServer server, IMessagePact messagePact, PactSpecification version)
+        {
+            PactHandle handle = server.NewPact(messagePact.Consumer, messagePact.Provider);
+            server.WithSpecification(handle, version);
+
+            return server.NewMessagePact(messagePact.Consumer, messagePact.Provider);
```

Comment:
Hang on - it creates a request/response pact and then immediately creates a new message pact? I'm not sure that's what you meant to do. Also the request builder class above also called `NewMessagePact` so it'll effectively overwrite this anyway.